frontend/workflows-lib/lib/types.ts

```diff
@@ -8,3 +8,17 @@ export interface Task {
 export interface TaskNode extends Task {
   children?: TaskNode[];
 }
+
+export enum WorkflowStatus {
+  Pending = "Pending",
+  Running = "Running",
+  Succeeded = "Succeeded",
+  Failed = "Failed",
+  Errored = "Errored",
+}
```

Comment:
Would it be better to use a union type (e.g. `type WorkflowStatus = "Pending" | "Running"`) instead?